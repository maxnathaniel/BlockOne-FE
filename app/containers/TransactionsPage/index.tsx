import React, { memo } from 'react';
import { useSelector } from 'react-redux';
import numeral from 'numeral';

import Table from '../../components/Table';

import {
  AccountName,
  Balance,
  TransactionsContainer,
  TransactionsInfo,
  TransactionsWrapper
} from './style';

const Transactions: React.FC<{}> = memo(() => {
  const { accounts, userConfig: { selectedAccount } } = useSelector((state: any) => state);
  const selectedAccountBalance = accounts[selectedAccount].amount;
  const transactions = accounts[selectedAccount].transactions;

  return (
    <TransactionsContainer>
      <TransactionsInfo>
        <AccountName>{selectedAccount}</AccountName>
        <Balance>{numeral(selectedAccountBalance).format('$0,0.00')}</Balance>
      </TransactionsInfo>
      <TransactionsWrapper>
        <Table data={transactions} />
      </TransactionsWrapper>
    </TransactionsContainer>
  );
});

export default Transactions;