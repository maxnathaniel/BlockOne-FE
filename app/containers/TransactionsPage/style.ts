import styled from '../../styles/styled-components';

interface ITransactionsContainerProps {
  position: string;
  height: string;
  width: string;
}

export const TransactionsContainer: any = styled.div<ITransactionsContainerProps>`
  position: absolute;
  right: 0;
  height: 100%;
  width: 85%;
  background-color: #192037;
  overflow: hidden;

  @media all and (max-width: 800px) {
    width: 100%;
  }
`;

export const TransactionsInfo = styled.div`
  display: flex;
  flex-flow: row wrap;
  font-weight: bold;
  text-align: center;
  margin: 20px auto;
  background-color: #272F4D;
  width: 95%;
  min-height: 140px;
  max-height: 200px;
  align-items: center;

  > * {
    flex: 1 100%;
  }
`;

export const AccountName = styled.h2`
  flex: 1;
  color: #FFF;
  margin: 0 !important;

  @media all and (max-width: 600px) {
    flex: 1 100%;
  }
`;

export const Balance = styled.h3`
  flex: 1;
  color: #FFF;
  margin: 0 !important;

  @media all and (max-width: 600px) {
    flex: 1 100%;
  }
`;

export const TransactionsWrapper = styled.div`
  min-height: 1200px;
  width: 95%;
  background-color: #272F4D;
  border-top: 2px solid #E94F90;
  margin: 10px auto;
`;