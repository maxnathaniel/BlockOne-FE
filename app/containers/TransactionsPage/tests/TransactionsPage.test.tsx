import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import TransactionsPage from '../index';

const accountId = '13535hr-35afa2-25252fa-26396ahb';
const defaultState = {
  accounts: {
    'Block One Savings': { amount: 100000, id: accountId, type: 'Savings', isSelected: true, transactions: [] },
  },
  userConfig: { selectedAccount: 'Block One Savings' }
};

jest.mock('react-redux', () => ({
  useSelector: () => defaultState,
  useDispatch: () => jest.fn(),
}));

const mock = require('react-redux');

describe('<TransactionsPage />', () => {
  it('should render the various inner Elements', () => {
    const { container, getByText } = render(<TransactionsPage />);
    expect(getByText('Block One Savings')).toBeInTheDocument();
    expect(getByText('$100,000.00')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});