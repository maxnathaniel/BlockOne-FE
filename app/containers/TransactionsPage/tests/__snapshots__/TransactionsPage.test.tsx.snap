// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionsPage /> should render the various inner Elements 1`] = `
<div>
  <div
    class="sc-AykKH qJuqI"
  >
    <div
      class="sc-AykKI fPmKgx"
    >
      <h2
        class="sc-AykKJ cYSRWo"
      >
        Block One Savings
      </h2>
      <h3
        class="sc-AykKK fWKuqF"
      >
        $100,000.00
      </h3>
    </div>
    <div
      class="sc-fzXfLO eyrABY"
    >
      <div
        class="sc-AykKC iLtsOi"
      >
        <div
          class="sc-AykKD dOvTbI"
        >
          <p
            class="sc-AykKE PnnHt"
          >
            Timestamp
          </p>
          <p
            class="sc-AykKE PnnHt"
          >
            Action
          </p>
          <p
            class="sc-AykKE PnnHt"
          >
            Description
          </p>
          <p
            class="sc-AykKE PnnHt"
          >
            Amount
          </p>
          <p
            class="sc-AykKE PnnHt"
          >
            Currency
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
